llama_index/storage/docstore/keyval_docstore.py

```diff
@@ -150,6 +150,91 @@ def add_documents(
                         node_key, metadata, collection=self._metadata_collection
                     )
 
+    async def aadd_documents(
+        self,
+        nodes: Sequence[BaseNode],
+        allow_update: bool = True,
+        batch_size: Optional[int] = None,
+        store_text: bool = True,
+    ) -> None:
+        """Add a document to the store.
+
+        Args:
+            docs (List[BaseDocument]): documents
+            allow_update (bool): allow update of docstore from document
+
+        """
+        batch_size = batch_size or self._batch_size
+        if batch_size > 1:
```

Comment:
I'm realizing now that batch size isn't actually doing anything here 😅 It's not being using by the underlying kvstore... oof.

I also had an idea for a refactor here -- will finish it in the morning :)